srtcore/buffer.cpp

```diff
@@ -538,7 +546,7 @@ int CSndBuffer::readData(const int offset, CPacket& w_packet, steady_clock::time
     // if found block is stale
     // (This is for messages that have declared TTL - messages that fail to be sent
     // before the TTL defined time comes, will be dropped).
-    if ((p->m_iTTL >= 0) && (count_milliseconds(steady_clock::now() - p->m_tsOriginTime) > p->m_iTTL))
+    if ((p->m_iTTL >= 0) && (count_milliseconds(steady_clock::now() - p->m_tsTimestamp) > p->m_iTTL))
```

Comment:
You haven't mentioned that you're going to merge the Origin Time and Source Time fields, while these fields have been separated for a reason. In this place it was Origin Time used, that is, the time when the packet sending request has happened, regardless whether the Source Time was provided or not, and it wasn't intended to use Source Time instead of Origin Time here, if Source Time was provided. Did you really want to change this behavior?

Reply:
> Did you really want to change this behavior?

I did. Please see the motivation in the description of this PR.

> This PR removes the redundant timestamp. Now sender buffer block has only one timestamp `m_tsTimestamp`, holding either source time if provided, or packet submission time.

Reply:
Ok, I missed that. Still, this concerns the behavior of Sender TL Packet Drop, that is, dropping packets that would be dropped anyway by the receiver. The problem is that this works together with the timestamp of the starting point, independently on the sender and receiver. The only thing that is required for the user-supplied timestamp is that they represent the time intervals between packets that should be maintained on the receiver side, with first ever received packet defining the initial value that is confronted with the connection start time to define the transmission start time being a base for conversion of the next timestamps. Because of that it has only a loose connection to the real packet scheduling time. Are you really sure that the sender-tlpktdrop facility will work correctly when you change the time representation of the sending time?

Note that we have still reported findings from users that suggest the sender dropping procedure happens to drop packets too eagerly. I wouldn't make any changes here until I'm certain that this at least contributes to fixing this problem. Also, separation of these fields isn't absolutely necessary to fix the time inaccuracy problem.